### ISSUE TITLE
Accessibility - Main Country List - Incorrect Reading Order

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Child-Labor-Android#32]
After reading what should be the heading, screen TalkBack reads the first country after the letter ‘A’ and then the letter ‘A’ followed by the next country. Same thing happens with the letter ‘B’ too, and I’m assuming the same for other sections on the screen. A similar thing happens for the ‘By Level’ and ‘By region’ screen where TalkBack reads first country then the section title (should be a heading) and then the next country.

The problem is for all the different tabs on this screen (A –Z, By Level, and By Region).

**Comment by aquilent-dgemoets** [comment]
Default Android List views do not support unselectable "heading rows," so to meet this requirement, a third-party custom list view was used.  Unfortunately, this list view was not optimized for accessibility, and based on initial analysis, it would be difficult to correct its reading by TalkBack (also applies to #41).

**Comment by aquilent-dgemoets** [comment]
We've discussed an alternative UI in which we remove the (inaccessible) third party custom list view, and replace the headings with an accessible dropdown menu at the top of the list.  The dropdown menu would filter the list to include only the selected countries (e.g., all of the "Significant Advancement" countries).  The A-Z list would neither have a dropdown nor a heading for each letter.

This approach will need to be estimated by the developer, approved by usability testers, triaged along with other tasks, and implemented by the developer.

**Comment by jfc3-dol** [comment]
Why can't a view be made with a heading of the letter the list of countries under it start with such as 

### A
Afghanistan
Albania
Algeria
Andorra
...

Having a drop down list would be difficult/confusing to use by those with disabilities and those without disabilities.

**Comment by aquilent-dgemoets** [comment]
I might not be understanding your suggestion, but it sounds like you're describing the current view that has a heading with the letter, then the countries with that letter-- please correct me if I'm wrong.  Unfortunately, this third-party view has accessibility problems, so we've devised an alternative using standard Android views.

I agree that the drop-down is less usable (for those with disabilities and those without), but it addresses the accessibility issue with the third-party view.

**Comment by jfc3-dol** [comment]
That is what I mean, but wouldn't think it's that difficult to create or that it would be a standard view from Android.

**Comment by jfc3-dol** [comment]
I guess, since you removed/changed third party widgets to create the country lists in the different sections it makes it a better experience but would be better having headings on the different screen/tabs.

**Comment by jfc3-dol** [comment]
Closing issue.
